### PR TITLE
Fixes #779: Specify jax cpu and update installation instructions

### LINKED ISSUE #779: update build instructions
```
conda env create -f environment.yml -n timemachine
conda install openmm=7.5.1 -c conda-forge # only if using openmm from conda
conda activate timemachine
```

should probably be:

```
conda env create -f environment.yml -n timemachine
conda activate timemachine
conda install openmm=7.5.1 -c conda-forge # only if using openmm from conda
```

else openmm gets installed into `base` env

and

we should probably add mypy to the `requirements.txt` even if we're not installing in developer mode since it provides the `stubgen` needed for `make` to succeed

### PR DESCRIPTION
**Pull request by badisa**
fixes #779

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,8 +39,8 @@ If using conda the following can be used to configure your environment
 
 ```shell
 conda env create -f environment.yml -n timemachine
-conda install openmm=7.5.1 -c conda-forge # only if using openmm from conda
 conda activate timemachine
+conda install openmm=7.5.1 -c conda-forge # only if using openmm from conda
```
Comment (mcwitt) [inline README.md:43]: ah, good catch!

File: setup.py [hunk 2]
```diff
@@ -92,6 +92,7 @@ def build_extension(self, ext):
         "pymbar>3.0.4",
         "pyyaml",
         "scipy",
+        "mypy",
```
Comment (mcwitt) [inline setup.py:95]: Could you comment on why this change was needed?

File: environment.yml [hunk 3]
```diff
@@ -5,6 +5,6 @@ channels:
 dependencies:
   - python=3.7
   - jax=0.3.14
-  - jaxlib=0.3.14
+  - jaxlib=0.3.14=*cpu*
```
Comment (mcwitt) [inline environment.yml:8]: nit: maybe
```suggestion
  - jaxlib=0.3.14=cpu*
```
(since `cpu`/`cuda` is at the beginning of the build description string)

File: setup.py (hunk 2 above)
Comment (badisa) [reply setup.py:95]: Because of the change to require `stubgen` to build the package, you can't install this without mypy based on this comment (https://github.com/proteneer/timemachine/pull/711#discussion_r848956244)

We could go back to an explicit version of stubgen and keep mypy as a separate requirement. We probably should add some form of testing in CI that doing `pip install .` works out of the box, as right now it doesn't.

Comment (badisa) [reply setup.py:95]: After digging in a bit more with @mcwitt it looks like we have a discontinuity between `make build` and `pip install`. `pip install` reaches into `pyproject.toml` and gets the build requirements there (that include mypy) and builds just fine, however `make build` will fail as `mypy` isn't in the env. 

Probably warrants an offline chat on how to proceed

Comment (mcwitt) [reply setup.py:95]: Just a note that if we decide to leave things as-is, we should probably add a note to the readme that `make build` only works after installing the dev dependencies (`pip install .[dev]`)

Comment (proteneer) [reply setup.py:95]: ah I see! I'm okay either way. (I do make build)

File: environment.yml [hunk 4]
```diff
@@ -5,6 +5,6 @@ channels:
 dependencies:
   - python=3.7
   - jax=0.3.14
-  - jaxlib=0.3.14
+  - jaxlib=0.3.14=cpu*
```
Comment (proteneer) [inline environment.yml:8]: just checking that this should be `=cpu*` and not `=*cpu*`, if the former works that's probably preferable?

Comment (mcwitt) [reply environment.yml:8]: I suggested the change from `=*cpu*` to `=cpu*`, but only had a very minor preference (`cpu` is at the beginning of the string in the current schema; I think it comes down to whether we want things to break if they change the schema or continue to work (but slight chance of doing something unexpected))

**Review by jkausrelay — APPROVED** [review]

**Review by proteneer — APPROVED** [review]
lgtm

**Review by mcwitt — APPROVED** [review]
lgtm!
